pgxn/neon/walproposer.c

```diff
@@ -1024,7 +1024,8 @@ DetermineEpochStartLsn(WalProposer *wp)
 	dth = &wp->safekeeper[wp->donor].voteResponse.termHistory;
 	wp->propTermHistory.n_entries = dth->n_entries + 1;
 	wp->propTermHistory.entries = palloc(sizeof(TermSwitchEntry) * wp->propTermHistory.n_entries);
-	memcpy(wp->propTermHistory.entries, dth->entries, sizeof(TermSwitchEntry) * dth->n_entries);
+	if (dth->n_entries > 0)
```

Comment:
Why this is needed? copying 0 bytes is valid, no?

Reply:
I propose this change based on commit https://github.com/neondatabase/postgres/commit/46ab07ffda9d6c8e63360ded2d4568aa160a7700

Tom Lane said in the commit message that this is prohibited by POSIX.

Reply:
But pointer here should never be NULL. Ok, I was just wondering.

Reply:
I think, I can demonstrate reaching that code with dth->entries == null, if it could be of interest.